omegaconf/basecontainer.py

```diff
@@ -769,3 +778,30 @@ def _update_types(node: Node, ref_type: type, object_type: Optional[type]) -> No
         if new_ref_type is not Any:
             node._metadata.ref_type = new_ref_type
             node._metadata.optional = new_is_optional
+
+
+def _instantiate_structured_config_impl(
+    conf: "DictConfig", instance_data: Dict[str, Any]
+) -> Any:
+    """Instantiate an instance of `conf._metadata.object_type`, populated by `instance_data`."""
+    from ._utils import get_structured_config_field_names
+
+    object_type = conf._metadata.object_type
+
+    object_type_field_names = get_structured_config_field_names(object_type)
```

Comment:
Can you make it a set here? This will speed up the `in` checks below in case of a large dataclass.

Reply:
Done in https://github.com/omry/omegaconf/pull/502/commits/6b014aea0ad0185722942a1315e6af22b2a9bdfe.

Reply:
And a related commit here: https://github.com/omry/omegaconf/pull/502/commits/4cefc6d859546949f65917b98ee3f36166f2104c.